ui/src/app/subnets/views/FormActions/components/AddFabric.tsx

```diff
@@ -0,0 +1,54 @@
+import { Input } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import type { FormActionProps } from "../FormActions";
+
+import FormikField from "app/base/components/FormikField";
+import FormikForm from "app/base/components/FormikForm";
+import { actions as fabricActions } from "app/store/fabric";
+import fabricSelectors from "app/store/fabric/selectors";
+
+type AddFabricValues = {
+  name: string;
+};
+
+const AddFabric = ({ activeForm, setActiveForm }: FormActionProps) => {
+  const dispatch = useDispatch();
+  const isSaving = useSelector(fabricSelectors.saving);
+  const isSaved = useSelector(fabricSelectors.saved);
+  const errors = useSelector(fabricSelectors.errors);
+
+  return (
+    <FormikForm<AddFabricValues>
+      buttonsBordered
+      allowAllEmpty
+      initialValues={{ name: "" }}
+      onSaveAnalytics={{
+        action: "Add fabric",
+        category: "Subnets form actions",
+        label: "Add fabric",
+      }}
+      submitLabel={`Add ${activeForm}`}
+      onSubmit={({ name }: { name: string }) => {
+        dispatch(fabricActions.create({ name }));
+      }}
+      onCancel={() => setActiveForm(undefined)}
+      resetOnSave
```

Comment:
The pattern we generally use for header forms is to close them when successfully submitted. For that we have an `onSuccess` prop that runs if `saved` changes from `false` to `true`, so this could be replaced with:
```
onSuccess={() => setActiveForm(null)}
```
Some other forms will include a secondary submit button with the label "Save and add another" (e.g. add machine) for cases where you might want to submit the form multiple times.

Reply:
Gotcha.

This will actually help mitigate an issue that I noticed:

[FormikFormButtons](https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/base/components/FormikFormButtons/FormikFormButtons.tsx) disabled attribute is not in sync with its visual indicators for loading/success state.

This makes for a slightly confusing U, but most importantly makes it possible to make multiple requests unintentionally (e.g. user thinks nothing has happened and needs to try again).

Let me know if I used the component incorrectly, but even if I did, this shoul be looked at - I think that we should make such inconsistent state impossible.

You can see what I mean for yourself if you spin up maas-ui from a commit before I made this change: https://github.com/canonical-web-and-design/maas-ui/pull/3504/commits/a5c6d6385645323c63ea8804343c1cb1b30f61ce and press the Add Fabric submit button multiple times.

Reply:
> If you mean that the button _should_ be disabled when the form is saving/saved but isn't

Yeah, that's what I meant.